contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,1029 @@
+            return fbr.getValue();
+        }
+    }
+
+    /**
+     * Enum of values used for {@link CFNumberType} in {@link #CFNumberGetValue} and
+     * {@link #CFNumberGetType}. Use {@link CFNumberType#typeIndex} for the expected
+     * integer value corresponding to the C-style enum.
+     */
+    enum CFNumberType {
+        unusedZero, kCFNumberSInt8Type, kCFNumberSInt16Type, kCFNumberSInt32Type, kCFNumberSInt64Type,
+        kCFNumberFloat32Type, kCFNumberFloat64Type, kCFNumberCharType, kCFNumberShortType, kCFNumberIntType,
+        kCFNumberLongType, kCFNumberLongLongType, kCFNumberFloatType, kCFNumberDoubleType, kCFNumberCFIndexType,
+        kCFNumberNSIntegerType, kCFNumberCGFloatType, kCFNumberMaxType;
+
+        /**
+         * Index for the type of {@link CFNumberRef} stored.
+         *
+         * @return a {@link CFIndex} representing the enum ordinal.
+         */
+        public CFIndex typeIndex() {
+            return new CFIndex(this.ordinal());
+        }
+    }
+
+    /**
+     * A reference to a {@code CFBoolean} object.
+     */
+    class CFBooleanRef extends CFTypeRef {
+        public CFBooleanRef() {
+            super();
+        }
+
+        public CFBooleanRef(Pointer p) {
+            super(p);
+            if (!isTypeID(BOOLEAN_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFBoolean. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convert a reference to a Core Foundations Boolean into its {@code boolean}
+         *
+         * @return The corresponding {@code boolean}
+         */
+        public boolean booleanValue() {
+            return 0 != INSTANCE.CFBooleanGetValue(this);
+        }
+    }
+
+    /**
+     * A reference to an immutable {@code CFArray} object.
+     * <p>
+     * CFArray is “toll-free bridged” with its Cocoa Foundation counterpart,
+     * {@code NSArray}. Therefore, in a method where you see an {@code NSArray *}
+     * parameter, you can pass in a {@link #CFArrayRef}.
+     */
+    class CFArrayRef extends CFTypeRef {
+        public CFArrayRef() {
+            super();
+        }
+
+        public CFArrayRef(Pointer p) {
+            super(p);
+            if (!isTypeID(ARRAY_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFArray. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convenience method for {@link #CFArrayGetCount} on this object
+         *
+         * @return The number of values in this array.
+         */
+        public CFIndex getCount() {
+            return INSTANCE.CFArrayGetCount(this);
+        }
+
+        /**
+         * Convenience method for {@link #CFArrayGetValueAtIndex} on this object
+         *
+         * @param idx
+         *            The index of the value to retrieve.
+         * @return The value at the {@code idx} index.
+         */
+        public Pointer getValueAtIndex(CFIndex idx) {
```

Comment:
Would it make sense to accept an int/long here and do the CFIndex wrapping/releasing in the function?